### PR TITLE
Start edit scripts

### PR DESCRIPTION
**Pull request by johrstrom**
This starts support for editing scripts. This is about as far as I'm going to go in this PR.

To test this
* create a script that shows up in projects#show
* press the edit button for that script
* now you'll see an `Add new option` button.
* Press that button and you'll get a new in progress field as shown below
* `Add` does nothing at the moment, while `Remove` works.
* Changing the field to choose changes the help text.

image

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1135148780858012/1204383563053604) by [Unito](https://www.unito.io)

### CODE REVIEW

File: apps/dashboard/app/controllers/scripts_controller.rb [hunk 1]
```diff
@@ -25,11 +22,20 @@ def create
...
+  def save
+    redirect_to project_path(params[:project_id]), notice: 'TODO'
+  end
+
   # POST   /projects/:project_id/scripts/:id/submit
   # submit the job
   def submit
-    project = Project.find(params[:project_id])
-    @script = Script.find(params[:id], project.directory)
     opts = submit_script_params[:script].to_h.symbolize_keys
```
Comment (Oglopf) [inline apps/dashboard/app/controllers/scripts_controller.rb:39]: Did you want the `find_script` to replace these, or not worry about this for now?

**Review by Oglopf — COMMENTED** [review]
This all looks good and works, I just wanted to check you didn't want to include a return value in the `scripts_controller` for the `submit` with `find_script` and also the whole `show` is removed which I wasn't sure if you wanted or not.

File: apps/dashboard/app/controllers/scripts_controller.rb (hunk 1 above)
Comment (johrstrom) [reply apps/dashboard/app/controllers/scripts_controller.rb:39]: Yea I figure we can use `before_action` to get rid of some duplication.

**Review by Oglopf — DISMISSED** [review]
Looks great.

**Comment by johrstrom** [comment]
> This all looks good and works, I just wanted to check you didn't want to include a return value in the scripts_controller for the submit with find_script and also the whole show is removed which I wasn't sure if you wanted or not.

Yea, I think it's fine to remove `show`, rails will run the `before_action` and render the right page. It doesn't have anymore logic than what we do in `find_script`.

WRT the return value `submit` still has redirects and rails will take care of `show`, but you're right in spotting this: we could fail to find the project or the script if say you edit the URL to something funny. So I've added failure logic to `find_script` if someone happens to edit the URL by hand to a project or script that doesn't exist.

**Comment by johrstrom** [comment]
> This all looks good and works, I just wanted to check you didn't want to include a return value in the scripts_controller for the submit with find_script and also the whole show is removed which I wasn't sure if you wanted or not.

>> Yea, I think it's fine to remove show, rails will run the before_action and render the right page. It doesn't have anymore logic than what we do in find_script.

>> WRT the return value submit still has redirects and rails will take care of show, but you're right in spotting this: we could fail to find the project or the script if say you edit the URL to something funny. So I've added failure logic to find_script if someone happens to edit the URL by hand to a project or script that doesn't exist.

Though I'm now seeing `scripts#new` could fail in the same manner, so I'll file a ticket for the same.

**Comment by Oglopf** [comment]
Sounds good, I'll approve after the tests finish.

**Review by Oglopf — APPROVED** [review]
LGTM.

**Pull request merged**